config/config.go

```diff
@@ -363,6 +366,10 @@ func (c *Cluster) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return fmt.Errorf("`cluster.heartbeat` cannot be unset for %q", c.Name)
 	}
 
+	if strings.HasSuffix(c.ClusterUsers[0].Name, "_*") && len(c.HeartBeat.Name) == 0 {
```

Comment:
Regarding hearbeats we recently did a change to use the /ping endpoint of clickhouse so that, by default (if the user doesn't change the request string), chproxy doesn't need a user/pwd anymore to send hearbeats to clickhouse https://github.com/ContentSquare/chproxy/pull/214.

so you should change your conditions (to check is it's the '/ping or not) & modify all the comments saying that by default the first user in clusters.users will be used